### PR TITLE
[enhancement](memtracker)  Use proc/meminfo MemAvailable to control memory and optimize MemTracker log printing

### PR DESCRIPTION
**Pull request by xinyiZzz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. When Doris BE is deployed with Hadoop, there is an OOM risk in only using BE process memory control limit at present, and additionally use `/proc/meminfo MemAvailable` to control memory;
2. Optimize the log printing of MemTracker, and fix the bug that Jemalloc does not print the log when the process memory exceeds the limit.
3. Print query peak mem usage in `fe.audit.log`.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/mem_info.cpp [hunk 1]
```diff
@@ -115,15 +117,49 @@ void MemInfo::init() {
...
-              << ", origin config value: " << config::mem_limit;
+              << ", origin config value: " << config::mem_limit
+              << ", System Mem Available Min Reserve: "
+              << PrettyPrinter::print(_s_sys_mem_available_low_water_mark, TUnit::BYTES)
+              << ", Vm Min Free KBytes: "
+              << PrettyPrinter::print(_s_vm_min_free_kbytes, TUnit::BYTES);
     _s_initialized = true;
 }
 #else
+MemInfo::refresh_proc_meminfo() {};
```
Comment (adonis0147) [inline be/src/util/mem_info.cpp:161]:
```suggestion
void MemInfo::refresh_proc_meminfo() {}
```

**Review by adonis0147 — CHANGES_REQUESTED** [review]

File: be/src/util/mem_info.cpp (hunk 1 above)
Comment (xinyiZzz) [reply be/src/util/mem_info.cpp]: fix, thks~

**Review by adonis0147 — APPROVED** [review]
LGTM

**Review by dataroaring — COMMENTED** [review]
LGTM

**Pull request merged**